crates/router/src/core/routing.rs

```diff
@@ -909,26 +909,43 @@ pub async fn retrieve_default_fallback_algorithm_for_profile(
 }
 
 #[cfg(feature = "v1")]
-
 pub async fn retrieve_default_routing_config(
     state: SessionState,
+    profile_id: Option<common_utils::id_type::ProfileId>,
     merchant_account: domain::MerchantAccount,
     transaction_type: &enums::TransactionType,
 ) -> RouterResponse<Vec<routing_types::RoutableConnectorChoice>> {
     metrics::ROUTING_RETRIEVE_DEFAULT_CONFIG.add(&metrics::CONTEXT, 1, &[]);
     let db = state.store.as_ref();
-
-    helpers::get_merchant_default_config(
-        db,
-        merchant_account.get_id().get_string_repr(),
-        transaction_type,
-    )
-    .await
-    .map(|conn_choice| {
-        metrics::ROUTING_RETRIEVE_DEFAULT_CONFIG_SUCCESS_RESPONSE.add(&metrics::CONTEXT, 1, &[]);
-        service_api::ApplicationResponse::Json(conn_choice)
-    })
+    if let Some(profile_id) = profile_id {
+        helpers::get_merchant_default_config(db, profile_id.get_string_repr(), transaction_type)
+            .await
+            .map(|conn_choice| {
+                metrics::ROUTING_RETRIEVE_DEFAULT_CONFIG_SUCCESS_RESPONSE.add(
+                    &metrics::CONTEXT,
+                    1,
+                    &[],
+                );
+                service_api::ApplicationResponse::Json(conn_choice)
+            })
+    } else {
+        helpers::get_merchant_default_config(
+            db,
+            merchant_account.get_id().get_string_repr(),
```

Comment:
Why are we having support for merchant_id, I think we might not need that

Reply:
Can you check these two:
So basically we are doing this, to make dashboard backwards compatible.
https://github.com/juspay/hyperswitch/pull/6546#discussion_r1839852600
https://github.com/juspay/hyperswitch/pull/6546#discussion_r1839848057